### DISCUSSION TITLE
DeepSpeed zero3

### DISCUSSION THREAD

**Discussion by Abolfazl-kr** [ymcui/Chinese-LLaMA-Alpaca-2#463] [Q&A]
what should we do to use deep speed stage3?
what changes should apply on deep speed config and the codes?

----
Admin Note: Transfer to Discussion, as this is not an issue.

**Comment by ymcui** [comment]
Such general questions should refer to the documentation of DeepSpeed. 
We did not use ZeRO-3 when training our models so we cannot provide specific comments.

**Reply by Abolfazl-kr** [reply]
Ok. 
I'm struggling with overflow and minimizing loss-scale, and i couldn't use bp16 because of my hardware. i want to try zero-3 maybe mixed precision work.

Thanks a lot.

**Reply by feifei05** [reply]
ValueError: ZeRO inference only makes sense with ZeRO Stage 3 - please adjust your config.
Should I replace it with zero-3?

**Comment by feifei05** [comment]
![微信图片_20240102092551](https://github.com/ymcui/Chinese-LLaMA-Alpaca-2/assets/69673314/afbc6f8d-df39-4774-9a05-71444379335d)

**Comment by yuchengml** [comment]
@Abolfazl-kr  @feifei05 
Please refer to [sft_with_ds_zero3](https://github.com/yuchengml/Chinese-LLaMA-Alpaca-2/tree/sft_with_ds_zero3)
